src/test/java/itest/ReportIT.java

```diff
@@ -0,0 +1,160 @@
+            CompletableFuture<JsonObject> postResponse = new CompletableFuture<>();
+            MultiMap form = MultiMap.caseInsensitiveMultiMap();
+            form.add("recordingName", TEST_RECORDING_NAME);
+            form.add("duration", "1");
+            form.add("events", "template=ALL");
+
+            webClient
+                    .post(RECORDING_REQ_URL)
+                    .sendForm(
+                            form,
+                            ar -> {
+                                if (assertRequestStatus(ar, postResponse)) {
+                                    MatcherAssert.assertThat(
+                                            ar.result().statusCode(), Matchers.equalTo(201));
+                                    postResponse.complete(null);
+                                }
+                            });
+
+            postResponse.get();
+
+            // Save the recording to archive
+            webClient
+                    .patch(String.format("%s/%s", RECORDING_REQ_URL, TEST_RECORDING_NAME))
+                    .sendBuffer(
+                            Buffer.buffer("SAVE"),
+                            ar -> {
+                                if (assertRequestStatus(ar, savedRecordingName)) {
+                                    MatcherAssert.assertThat(
+                                            ar.result().statusCode(), Matchers.equalTo(200));
+                                    MatcherAssert.assertThat(
+                                            ar.result()
+                                                    .getHeader(HttpHeaders.CONTENT_TYPE.toString()),
+                                            Matchers.equalTo(HttpMimeType.PLAINTEXT.mime()));
+                                    savedRecordingName.complete(ar.result().bodyAsString());
+                                }
+                            });
+
+            savedRecordingName.get();
+
+            // Get a report for the above recording
+            CompletableFuture<Buffer> getResponse = new CompletableFuture<>();
+            webClient
+                    .get(String.format("%s/%s", REPORT_REQ_URL, savedRecordingName.get()))
+                    .send(
+                            ar -> {
+                                if (assertRequestStatus(ar, getResponse)) {
+                                    MatcherAssert.assertThat(
+                                            ar.result().statusCode(), Matchers.equalTo(200));
+                                    MatcherAssert.assertThat(
+                                            ar.result()
+                                                    .getHeader(HttpHeaders.CONTENT_TYPE.toString()),
+                                            Matchers.equalTo(HttpMimeType.HTML.mime()));
+                                    getResponse.complete(ar.result().bodyAsBuffer());
+                                }
+                            });
+
+            getResponse.get();
```

Comment:
Since I'm only checking the content-type header, what validations for the response body would be appropriate here?

Reply:
I found this `TODO` in `RecordingWorkflowIT`. I can parse the response body in both report itest files and `RecordingWorkflowIT` to verify the response is valid HTML.
```
// verify that reports can be downloaded successfully and yield non-empty HTML documents
            // (TODO: verify response body is a valid HTML document)
            String reportUrl = recordingInfo.getString("reportUrl");
            Path reportPath =
                    downloadFileAbs(reportUrl, TEST_RECORDING_NAME + "_report", ".html")
                            .get(REQUEST_TIMEOUT_SECONDS, TimeUnit.SECONDS);
            MatcherAssert.assertThat(reportPath.toFile().length(), Matchers.greaterThan(0L));
```

Reply:
> Since I'm only checking the content-type header, what validations for the response body would be appropriate here?

You could just complete the Future with the response header(s) and perform the assertions after/on the `getResponse.get()`, if the purpose of this test is just to check the Content-Type.

It might make sense to use JSoup to do other assertions about the response however, beyond just checking the Content-Type header. It doesn't need to be and probably shouldn't be too in-depth, since the exact format of the rules analysis document can change between JMC versions, but some basic checks that the received content really is a parseable HTML document with some general overall structure would be fine. Very similar to what the ClientAssetsIT does.